modules/oe_subscriptions_anonymous/tests/src/Functional/TermsConfigurationTest.php

```diff
@@ -0,0 +1,137 @@
+    $article = $this->drupalCreateNode([
+      'type' => 'article',
+      'status' => 1,
+    ]);
+    $page = $this->drupalCreateNode([
+      'type' => 'page',
+      'status' => 1,
+    ]);
+
+    $assert_session = $this->assertSession();
+
+    // Assert that the link is not present if the configuration is not set.
+    $this->drupalGet(Url::fromRoute('oe_subscriptions_anonymous.subscription_request', [
+      'flag' => 'subscribe_all',
+      'entity_id' => $article->id(),
+    ]));
+    $assert_session->fieldExists('I have read and agree with the data protection terms.');
+    $assert_session->linkNotExists('data protection terms');
+
+    // User without permission can't access the configuration page.
+    $this->drupalGet(Url::fromRoute('oe_subscriptions_anonymous.settings'));
+    $assert_session->pageTextContains('You are not authorized to access this page.');
+    $assert_session->statusCodeEquals(403);
+
+    // Assert user with permissions can manage configuration.
+    $user = $this->createUser(['administer anonymous subscriptions']);
+    $this->drupalLogin($user);
+    $this->drupalGet(Url::fromRoute('oe_subscriptions_anonymous.settings'));
+    $url_field = $assert_session->fieldExists('Terms page URL');
+    $url_field->setValue($page->label() . ' (' . $page->id() . ')');
+    $assert_session->buttonExists('Save configuration')->press();
+    $assert_session->statusMessageContains('The configuration options have been saved.');
+    $this->drupalLogout();
+
+    // The link is present in page.
+    $this->drupalGet(Url::fromRoute('oe_subscriptions_anonymous.subscription_request', [
+      'flag' => 'subscribe_all',
+      'entity_id' => $page->id(),
+    ]));
+    $this->clickLink('data protection terms');
+    $assert_session->addressEquals($page->toUrl());
+
+    // The link is present in article and it's the same than in page.
+    $this->drupalGet(Url::fromRoute('oe_subscriptions_anonymous.subscription_request', [
+      'flag' => 'subscribe_all',
+      'entity_id' => $article->id(),
+    ]));
+    $this->clickLink('data protection terms');
+    $assert_session->addressEquals($page->toUrl());
+
+    // Delete node and check that the field is not present.
+    $page->delete();
+    $this->drupalGet(Url::fromRoute('oe_subscriptions_anonymous.subscription_request', [
+      'flag' => 'subscribe_all',
+      'entity_id' => $article->id(),
+    ]));
+    $assert_session->fieldExists('I have read and agree with the data protection terms.');
+    $assert_session->linkNotExists('data protection terms');
+
+    // Set external URL for terms page.
+    $this->drupalLogin($user);
+    $this->drupalGet(Url::fromRoute('oe_subscriptions_anonymous.settings'));
+    $url_field = $assert_session->fieldExists('Terms page URL');
+    $url_field->setValue('https://www.drupal.org/');
+    $assert_session->buttonExists('Save configuration')->press();
+    $assert_session->statusMessageContains('The configuration options have been saved.');
```

Comment:
Always add the message type, like in other tests (https://github.com/openeuropa/oe_subscriptions/blob/1.x/modules/oe_subscriptions_anonymous/tests/src/Functional/SubscribeTest.php#L97).

Reply:
There are no negative tests, entering an invalid link.

Reply:
Done